utils.go

```diff
@@ -35,3 +48,87 @@ func getFileContents(file string) (string, error) {
 
 	return string(bytes), nil
 }
+
+func getKernelVersion() (string, error) {
+	contents, err := getFileContents(procVersion)
+	if err != nil {
+		return "", err
+	}
+
+	fields := strings.Fields(contents)
+
+	if len(fields) < 3 {
+		return "", fmt.Errorf("unexpected contents in %v", procVersion)
+	}
+
+	version := fields[2]
+
+	if version == "" {
+		return "", fmt.Errorf("unexpected contents in %v", procVersion)
+	}
+
+	return version, nil
+}
+
+func getDistroDetails() (name, version string, err error) {
+	files := []string{osRelease, osReleaseClr}
+
+	for _, file := range files {
+
+		contents, err := getFileContents(file)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+
+			return "", "", err
+		}
+
+		lines := strings.Split(contents, "\n")
```

Comment:
I don't think a follow-up is necessary, not precisely a performance critical piece of code :)